src/lib/y2firewall/proposal_settings.rb

```diff
@@ -1,11 +1,44 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
 require "yast"
 
 module Y2Firewall
+  # Class that stores the proposal settings for firewalld during installation.
   class ProposalSettings
     include Yast::Logger
     include Yast::I18n
-    attr_accessor :enable_firewall, :enable_sshd, :open_ssh, :open_vnc
 
+    # [Boolean] Whether the firewalld service will be enable
+    attr_accessor :enable_firewall
```

Comment:
as you are having more rich enable/disable variants, does it make sense to still have public `enable_firewall=`?

Reply:
Probably not make sense all, is preferable to not expose it, but currently it is used in `load_feature` although I could just modify the `public_send` by `an instance_variable_set` in bang methods and also in some tests.

Reply:
Will try to replace in next iteration over code.